src/main/scala/com/gu/liveappversions/S3Storage.scala

```diff
@@ -16,62 +17,71 @@ import scala.util.{ Failure, Success, Try }
 
 object S3Storage {
 
-  private val s3Client = AmazonS3ClientBuilder.standard()
-    .withCredentials(Aws.credentials("developerPlayground"))
-    .withRegion(Aws.euWest1.getName)
+  private val s3Client = S3Client
+    .builder()
+    .credentialsProvider(Aws.credentials("developerPlayground"))
+    .region(Aws.euWest1)
     .build()
 
   def fullKey(env: Env, partialKey: String): String = {
     val stagePrefix = if (env.stage == "PROD") "/" else s"/${env.stage}/"
     s"reserved-paths$stagePrefix$partialKey"
   }
 
-  def putJson(jsonToUpload: Json, env: Env, bucketName: String, partialKey: String): Try[PutObjectResult] = {
-
+  def putJson(jsonToUpload: Json, env: Env, bucketName: String, partialKey: String): Try[PutObjectResponse] = {
     val key = fullKey(env, partialKey)
+    val accessControl = if (env.stage == "PROD") ObjectCannedACL.PUBLIC_READ else ObjectCannedACL.PRIVATE
+    val bytes = jsonToUpload.toString().getBytes(StandardCharsets.UTF_8)
+    val stream = new ByteArrayInputStream(bytes)
+
+    val putRequest = PutObjectRequest.builder()
+      .bucket(bucketName)
+      .key(key)
+      .contentType("application/json")
+      .cacheControl("max-age=60")
+      .acl(accessControl)
+      .build()
 
-    val buildAttributesStream: ByteArrayInputStream = new ByteArrayInputStream(jsonToUpload.toString().getBytes)
-
-    val metadata = new ObjectMetadata()
-    metadata.setContentType("application/json")
-    metadata.setCacheControl("max-age=60")
-
-    val accessControl = if (env.stage == "PROD") {
-      CannedAccessControlList.PublicRead
-    } else {
-      CannedAccessControlList.Private //It's preferable to avoid serving test files via https://mobile.guardianapis.com/
-    }
-
-    val putObjectRequest = new PutObjectRequest(bucketName, key, buildAttributesStream, metadata)
-      .withCannedAcl(accessControl)
-
-    Try(s3Client.putObject(putObjectRequest)) match {
+    Try {
+      val requestBody = RequestBody.fromInputStream(stream, bytes.length)
```

Comment:
Why do we need this in the try statement?

Reply:
Yeah this variable can either be moved up, or used inline in the `putObject` method